File: Applied Course/4.Problem Solving/1.Problems on Array/6.Sort Array By Parity.py
# https://leetcode.com/problems/sort-array-by-parity/
# Logic-1
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        return [i for i in A if i%2==0]+[i for i in A if i%2!=0]
# Logic-2
class Solution:
    def sortArrayByParity(self, A: List[int]) -> List[int]:
        i=0
        j=len(A)-1
        while(i<j):
            if A[i]%2!=0:
                A[i],A[j]=A[j],A[i]
                j-=1
            else:
                i+=1
        return A

File: Applied Course/4.Problem Solving/1.Problems on Array/6.Sort Array By Parity.cpp
#include<bits/stdc++.h>
using namespace std;
// Logic-1
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        vector<int> ans;
        for(auto ele:A)
            if(ele%2==0)
                ans.push_back(ele);
        for(auto ele:A)
            if(ele%2!=0)
                ans.push_back(ele);
        return ans;
    }
};
// Logic-2
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int i=0,j=A.size()-1;
        while(i<j)
        {
            if(A[i]%2!=0)
            {
                swap(A[i],A[j]);
                j--;
            }
            else
                i++;
        }
        return A;
    }
};
// Logic-3
int main()
{
    // https://leetcode.com/problems/sort-array-by-parity/  
    return 0;
}